### PR TITLE
fluid_nav_bar, gooey_edge and bubble_tab_bar nnbd fixed

### PR DESCRIPTION
**Pull request by SoroushBeigi**

### CODE REVIEW

File: vignettes/fluid_nav_bar/lib/fluid_nav_bar.dart [hunk 1]
```diff
@@ -163,7 +148,7 @@ class _FluidNavBarState extends State<FluidNavBar> with TickerProviderStateMixin
     _yController.animateTo(0.0, duration: Duration(milliseconds: 300));
 
     if (widget.onChange != null) {
-      widget.onChange(index);
+      widget.onChange!(index);
     }
```
Comment (esDotDev) [inline vignettes/fluid_nav_bar/lib/fluid_nav_bar.dart:152]: You can call this in a null-safe way in one line like:
```
widget.onChange?.call(index);
```

File: vignettes/gooey_edge/lib/gooey_carousel.dart [hunk 2]
```diff
@@ -41,42 +40,47 @@ class GooeyCarouselState extends State<GooeyCarousel> with SingleTickerProviderS
...
+              isDragComplete: _dragCompleted,
+            )
+          ],
+        ));
   }
 
   Size _getSize() {
-    final RenderBox box = _key.currentContext.findRenderObject();
-    return box.size;
+    final RenderBox? box = _key.currentContext?.findRenderObject() as RenderBox;
```
Comment (esDotDev) [inline vignettes/gooey_edge/lib/gooey_carousel.dart:75]: Never cast away nullability.  `findRenderObject` returns `RenderObject?` so it must be cast as `RenderBox?` not `RenderBox`

File: vignettes/gooey_edge/lib/gooey_edge.dart [hunk 3]
```diff
@@ -14,30 +14,39 @@ class GooeyEdge {
...
+    } else if (side == Side.top) {
+      touchOffset = FractionalOffset(o.dy, 1.0 - o.dx);
+    } else {
+      touchOffset = FractionalOffset(1.0 - o.dy, o.dx);
+    }
   }
 
   Path buildPath(Size size, {double margin = 0.0}) {
-    if (points == null || points.length == 0) { return null; }
+    if (points.length == 0) {
```
Comment (esDotDev) [inline vignettes/gooey_edge/lib/gooey_edge.dart:47]: Minor nit, but `points.isEmpty` is preferred

File: vignettes/gooey_edge/lib/gooey_edge.dart [hunk 4]
```diff
@@ -70,40 +79,58 @@ class GooeyEdge {
   }
 
   void tick(Duration duration) {
-    if (points == null || points.length == 0) { return; }
+    if (points.length == 0) {
```
Comment (esDotDev) [inline vignettes/gooey_edge/lib/gooey_edge.dart:82]: `.isEmpty`

File: vignettes/gooey_edge/lib/sun_moon.dart [hunk 5]
```diff
@@ -69,7 +69,7 @@ class _SunAndMoonState extends State<SunAndMoon> with SingleTickerProviderStateM
     return Opacity(
       opacity: sin(currentAngle) < 0 ? 1 : 0,
       child: RotationTransition(
-        turns: _rotationAnimation,
+        turns: _rotationAnimation as Animation<double>,
```
Comment (esDotDev) [inline vignettes/gooey_edge/lib/sun_moon.dart:72]: Should be able to just declare the animation as `Animation<double>` up top?

**Review by esDotDev — CHANGES_REQUESTED** [review]

**Comment by SoroushBeigi** [comment]
bubble_tab_bar also finished

File: vignettes/bubble_tab_bar/lib/navbar_button.dart [hunk 6]
```diff
@@ -14,15 +14,15 @@ class NavbarButton extends StatefulWidget {
...
 
   @override
   _NavbarButtonState createState() => _NavbarButtonState();
 }
 
 class _NavbarButtonState extends State<NavbarButton> with SingleTickerProviderStateMixin {
-  AnimationController _iconAnimController;
-  bool _wasSelected;
+  late AnimationController _iconAnimController;
+  bool _wasSelected = false;
```
Comment (esDotDev) [inline vignettes/bubble_tab_bar/lib/navbar_button.dart:25]: I think this should probably be:
```
late bool _wasSelected = widget.isSelected;
```
Not sure how much it matters, but I think is technically more correct.

**Review by esDotDev — CHANGES_REQUESTED** [review]
One small nit, looks great!

File: vignettes/bubble_tab_bar/lib/navbar_button.dart (hunk 6 above)
Comment (esDotDev) [reply vignettes/bubble_tab_bar/lib/navbar_button.dart:25]: I'll just land this now and fix locally.

Comment (SoroushBeigi) [reply vignettes/bubble_tab_bar/lib/navbar_button.dart:25]: Yes it is probably more appropriate, thanks

**Pull request merged**